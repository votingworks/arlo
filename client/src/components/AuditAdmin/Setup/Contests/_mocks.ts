export const contestsInputMocks = {
  inputs: [
    { key: 'Contest Name', value: 'Contest Name' },
    { key: 'Name of Candidate/Choice 1', value: 'Choice One' },
    { key: 'Name of Candidate/Choice 2', value: 'Choice Two' },
    { key: 'Votes for Candidate/Choice 1', value: '10' },
    { key: 'Votes for Candidate/Choice 2', value: '20' },
    { key: 'Total Ballot Cards Cast for Contest', value: '30' },
  ],
  errorInputs: [
    { key: 'Contest Name', value: '', error: 'Required' },
    {
      key: 'Total Ballot Cards Cast for Contest',
      value: '',
      error:
        'Must be greater than or equal to the sum of votes for each candidate/choice',
    },
    {
      key: 'Total Ballot Cards Cast for Contest',
      value: 'test',
      error: 'Must be a number',
    },
    {
      key: 'Total Ballot Cards Cast for Contest',
      value: '-1',
      error: 'Must be a positive number',
    },
    {
      key: 'Total Ballot Cards Cast for Contest',
      value: '0.5',
      error: 'Must be an integer',
    },
    { key: 'Name of Candidate/Choice 1', value: '', error: 'Required' },
    { key: 'Name of Candidate/Choice 2', value: '', error: 'Required' },
    {
      key: 'Votes for Candidate/Choice 1',
      value: '',
      error: 'Required',
    },
    {
      key: 'Votes for Candidate/Choice 1',
      value: 'test',
      error: 'Must be a number',
    },
    {
      key: 'Votes for Candidate/Choice 1',
      value: '-1',
      error: 'Must be a positive number',
    },
    {
      key: 'Votes for Candidate/Choice 1',
      value: '0.5',
      error: 'Must be an integer',
    },
    {
      key: 'Votes for Candidate/Choice 2',
      value: '',
      error: 'Required',
    },
    {
      key: 'Votes for Candidate/Choice 2',
      value: 'test',
      error: 'Must be a number',
    },
    {
      key: 'Votes for Candidate/Choice 2',
      value: '-1',
      error: 'Must be a positive number',
    },
    {
      key: 'Votes for Candidate/Choice 2',
      value: '0.5',
      error: 'Must be an integer',
    },
  ],
}

export default contestsInputMocks
