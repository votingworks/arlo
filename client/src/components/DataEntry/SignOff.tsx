import React from 'react'
import { Formik, Form } from 'formik'
import { H1 } from '@blueprintjs/core'
import FormSection from '../Atoms/Form/FormSection'
import { LabelText, NameField } from './Atoms'
import FormButton from '../Atoms/Form/FormButton'
import { IAuditBoard } from '../../types'

export interface IProps {
  auditBoard: IAuditBoard
  submitSignoff: (memberNames: string[]) => void
}

const SignOff = ({ auditBoard, submitSignoff }: IProps) => {
  return (
    <div>
      <H1>{auditBoard.name}: Board Member Sign-off</H1>
      <p>
        Thank you for completing the ballots assigned to your Audit Board.
        Please sign your name below to confirm that all ballots were audited to
        the best of your ability and in accordance with the appropriate state
        statutes and guidelines.
      </p>
      <p>
        If another round of auditing is needed, you will be notified by election
        officials.
      </p>
      <Formik
        initialValues={auditBoard.members.map(() => '')}
        onSubmit={submitSignoff}
        render={({ values, handleSubmit }) => (
          <Form>
            {auditBoard.members.map((member, i) => (
              <FormSection
                key={member.name}
                label={`Audit Board Member: ${member.name}`}
              >
                <LabelText htmlFor={`[${i}]`}>Full Name</LabelText>
                <NameField name={`[${i}]`} />
              </FormSection>
            ))}
            <FormButton
              intent="primary"
              type="button"
              onClick={handleSubmit}
              disabled={auditBoard.members.some((_, i) => !values[i])}
            >
              Sign Off
            </FormButton>
          </Form>
        )}
      />
    </div>
  )
}

export default SignOff
