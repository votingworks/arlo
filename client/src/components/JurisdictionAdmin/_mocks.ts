import { IBatches } from './useBatchResults'
import { IRound } from '../AuditAdmin/useRoundsAuditAdmin'
import { FileProcessingStatus } from '../useCSV'
import {
  IFullHandTallyBatchResults,
  IFullHandTallyBatchResult,
} from './useFullHandTallyResults'
import { mocksOfType } from '../testUtilities'

export interface INullResultValues {
  [contestId: string]: {
    [choiceId: string]: null | string | number
  }
}

export const roundMocks = mocksOfType<IRound>()({
  incomplete: {
    id: 'round-1',
    roundNum: 1,
    startedAt: '2020-09-14T17:35:19.482Z',
    endedAt: null,
    isAuditComplete: false,
    needsFullHandTally: false,
    isFullHandTally: false,
    drawSampleTask: {
      status: FileProcessingStatus.PROCESSED,
      startedAt: '2020-09-14T17:35:19.482Z',
      completedAt: '2020-09-14T17:36:19.482Z',
      error: null,
    },
  },
  complete: {
    id: 'round-1',
    roundNum: 1,
    startedAt: '2020-09-14T17:35:19.482Z',
    endedAt: '2020-09-14T17:35:19.482Z',
    needsFullHandTally: false,
    isAuditComplete: true,
    isFullHandTally: false,
    drawSampleTask: {
      status: FileProcessingStatus.PROCESSED,
      startedAt: '2020-09-14T17:35:19.482Z',
      completedAt: '2020-09-14T17:36:19.482Z',
      error: null,
    },
  },
  fullHandTallyIncomplete: {
    id: 'round-1',
    roundNum: 1,
    startedAt: '2020-09-14T17:35:19.482Z',
    endedAt: null,
    isAuditComplete: false,
    needsFullHandTally: true,
    isFullHandTally: true,
    drawSampleTask: {
      status: FileProcessingStatus.PROCESSED,
      startedAt: '2020-09-14T17:35:19.482Z',
      completedAt: '2020-09-14T17:36:19.482Z',
      error: null,
    },
  },
})

export const resultsMocks = mocksOfType<INullResultValues>()({
  emptyInitial: {
    'contest-id-1': {
      'choice-id-1': null,
      'choice-id-2': null,
    },
    'contest-id-2': {
      'choice-id-3': null,
      'choice-id-4': null,
    },
  },
  complete: {
    'contest-id-1': {
      'choice-id-1': 1,
      'choice-id-2': 2,
    },
    'contest-id-2': {
      'choice-id-3': 1,
      'choice-id-4': 2,
    },
  },
})

export const batchResultsMocks = mocksOfType<INullResultValues>()({
  empty: {
    'batch-1': {
      'choice-id-1': null,
      'choice-id-2': null,
    },
    'batch-2': {
      'choice-id-1': null,
      'choice-id-2': null,
    },
    'batch-3': {
      'choice-id-1': null,
      'choice-id-2': null,
    },
  },
  complete: {
    'batch-1': {
      'choice-id-1': 1,
      'choice-id-2': 2,
    },
    'batch-2': {
      'choice-id-1': 1,
      'choice-id-2': 2,
    },
    'batch-3': {
      'choice-id-1': 1,
      'choice-id-2': 2,
    },
  },
})

export const batchesMocks = mocksOfType<IBatches>()({
  emptyInitial: {
    batches: [
      {
        id: 'batch-1',
        name: 'Batch One',
        numBallots: 100,
        auditBoard: null,
        resultTallySheets: [],
      },
      {
        id: 'batch-2',
        name: 'Batch Two',
        numBallots: 100,
        auditBoard: null,
        resultTallySheets: [],
      },
      {
        id: 'batch-3',
        name: 'Batch Three',
        numBallots: 100,
        auditBoard: null,
        resultTallySheets: [],
      },
    ],
    resultsFinalizedAt: null,
  },
  complete: {
    batches: [
      {
        id: 'batch-1',
        name: 'Batch One',
        numBallots: 100,
        auditBoard: {
          id: 'ab-1',
          name: 'Audit Board One',
        },
        resultTallySheets: [
          {
            name: 'Tally Sheet #1',
            results: {
              'choice-id-1': 1,
              'choice-id-2': 2,
            },
          },
        ],
      },
      {
        id: 'batch-2',
        name: 'Batch Two',
        numBallots: 100,
        auditBoard: {
          id: 'ab-1',
          name: 'Audit Board One',
        },
        resultTallySheets: [
          {
            name: 'Tally Sheet #1',
            results: {
              'choice-id-1': 0,
              'choice-id-2': 10,
            },
          },
        ],
      },
      {
        id: 'batch-3',
        name: 'Batch Three',
        numBallots: 100,
        auditBoard: {
          id: 'ab-1',
          name: 'Audit Board One',
        },
        resultTallySheets: [
          {
            name: 'Tally Sheet #1',
            results: {
              'choice-id-1': 2000,
              'choice-id-2': 20,
            },
          },
        ],
      },
    ],
    resultsFinalizedAt: '2020-09-14T17:35:19.482Z',
  },
})

export const fullHandTallyBatchResultMock = mocksOfType<
  IFullHandTallyBatchResults
>()({
  empty: {
    finalizedAt: '',
    results: [],
  },
  complete: {
    finalizedAt: '',
    results: [
      {
        batchName: 'Batch1',
        batchType: 'Other',
        choiceResults: {
          'choice-id-1': 10,
          'choice-id-2': 20,
        },
      },
    ],
  },
  completeWithMultipleBatch: {
    finalizedAt: '',
    results: [
      {
        batchName: 'Batch1',
        batchType: 'Other',
        choiceResults: {
          'choice-id-1': 5,
          'choice-id-2': 15,
        },
      },
      {
        batchName: 'Batch2',
        batchType: 'Provisional',
        choiceResults: {
          'choice-id-1': 5,
          'choice-id-2': 5,
        },
      },
    ],
  },
  updated: {
    finalizedAt: '',
    results: [
      {
        batchName: 'Batch12',
        batchType: 'Other',
        choiceResults: {
          'choice-id-1': 10,
          'choice-id-2': 20,
        },
      },
    ],
  },
  finalized: {
    finalizedAt: '2021-04-13T15:01:00.383031+00:00',
    results: [
      {
        batchName: 'Batch12',
        batchType: 'Other',
        choiceResults: {
          'choice-id-1': 10,
          'choice-id-2': 20,
        },
      },
    ],
  },
})

export const fullHandTallyBatchResultsMock = mocksOfType<
  IFullHandTallyBatchResult
>()({
  empty: {
    batchName: '',
    batchType: '',
    choiceResults: {},
  },
  complete: {
    batchName: 'Batch1',
    batchType: 'Other',
    choiceResults: {
      'choice-id-1': 10,
      'choice-id-2': 20,
    },
  },
  updated: {
    batchName: 'Batch12',
    batchType: 'Other',
    choiceResults: {
      'choice-id-1': 10,
      'choice-id-2': 20,
    },
  },
})
