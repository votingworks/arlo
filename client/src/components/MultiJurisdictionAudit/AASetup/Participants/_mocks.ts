import { readFileSync } from 'fs'
import { join } from 'path'

const jurisdictionFile = new File(
  [
    readFileSync(
      join(
        __dirname,
        '../../../../../public/sample_jurisdiction_filesheet.csv'
      ),
      'utf8'
    ),
  ],
  'jurisdictions.csv',
  { type: 'text/csv' }
)

export default jurisdictionFile
