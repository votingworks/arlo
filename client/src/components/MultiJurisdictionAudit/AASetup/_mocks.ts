import { ElementType } from '../../../types'
import { setupStages, stageTitles } from './index'
import { ISidebarMenuItem } from '../../Atoms/Sidebar'

const relativeStages = (
  stage: ElementType<typeof setupStages>,
  state: ISidebarMenuItem['state'] = 'live'
): {
  prevStage: ISidebarMenuItem
  nextStage: ISidebarMenuItem
  menuItems: ISidebarMenuItem[]
} => {
  const prevId = setupStages[setupStages.indexOf(stage) - 1]
  const prevStage: ISidebarMenuItem = {
    id: prevId,
    title: stageTitles[prevId],
    active: false,
    activate: jest.fn(),
    state,
  }
  const nextId = setupStages[setupStages.indexOf(stage) + 1]
  const nextStage: ISidebarMenuItem = {
    id: nextId,
    title: stageTitles[nextId],
    active: false,
    activate: jest.fn(),
    state,
  }
  const menuItems = setupStages.map((s: ElementType<typeof setupStages>) => ({
    id: s,
    title: stageTitles[s],
    active: s === stage,
    activate: jest.fn(),
    state,
  }))
  return { prevStage, nextStage, menuItems }
}

export default relativeStages
