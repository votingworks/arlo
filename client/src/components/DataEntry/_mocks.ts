import { IAuditBoard, BallotStatus, Interpretation } from '../../types'
import { contestMocks } from '../MultiJurisdictionAudit/Setup/Contests/_mocks'
import { IBallot } from './Ballot'

export const contest = contestMocks.filledTargeted.contests[0]

export const doneDummyBallots: { ballots: IBallot[] } = {
  ballots: [
    {
      id: 'ballot-id-1',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-2',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-3',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-4',
      batch: {
        name: '0003-04-Precinct 31 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 1965,
      status: BallotStatus.NOT_FOUND,
      interpretations: [],
    },
  ],
}

export const dummyBallots: { ballots: IBallot[] } = {
  ballots: [
    {
      id: 'ballot-id-1',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: null,
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [
        {
          contestId: contest.id,
          interpretation: Interpretation.VOTE,
          choiceIds: [contest.choices[0].id],
          comment: 'Good ballot',
        },
      ],
    },
    {
      id: 'ballot-id-2',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-3',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-4',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-2',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-5',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-2',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-6',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-2',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-7',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-3',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-8',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-3',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-9',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-3',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-10',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-4',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-11',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-4',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-12',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-4',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-13',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-5',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-14',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-5',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-15',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-5',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-16',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-6',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-17',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-6',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-18',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-6',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-19',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-7',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-20',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-7',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-21',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-7',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-22',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-8',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-23',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-8',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-24',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-8',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-25',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-9',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-26',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-9',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-27',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-9',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
  ],
}

export const dummyBoards = (): IAuditBoard[] => [
  {
    id: 'audit-board-1',
    name: 'Audit Board #1',
    jurisdictionId: 'jurisdiction-1',
    jurisdictionName: 'Jurisdiction 1',
    roundId: 'round-1',
    members: [
      {
        name: 'John Doe',
        affiliation: '',
      },
      {
        name: 'Jane Doe',
        affiliation: 'LIB',
      },
    ],
    signedOffAt: null,
  },
  {
    id: 'audit-board-2',
    name: 'Audit Board #2',
    jurisdictionId: 'jurisdiction-1',
    jurisdictionName: 'Jurisdiction 1',
    roundId: 'round-1',
    members: [],
    signedOffAt: null,
  },
]

export default {
  dummyBoards,
  dummyBallots,
}
