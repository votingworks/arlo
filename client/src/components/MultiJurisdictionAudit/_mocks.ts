import { contestMocks } from './AASetup/Contests/_mocks'
import { IFileInfo } from './useJurisdictions'
import {
  auditSettings,
  manifestFile,
  talliesFile,
} from './useSetupMenuItems/_mocks'
import { FileProcessingStatus } from './useSetupMenuItems/getJurisdictionFileStatus'
import { IAuditSettings } from '../../types'
import { IBallot } from './RoundManagement/useBallots'
import { IAuditBoard } from './useAuditBoards'
import { IRound } from './useRoundsAuditAdmin'

const manifestFormData: FormData = new FormData()
manifestFormData.append('manifest', manifestFile, manifestFile.name)
const talliesFormData: FormData = new FormData()
talliesFormData.append('batchTallies', talliesFile, talliesFile.name)

export const jaApiCalls = {
  getUser: {
    url: '/api/me',
    response: {
      type: 'jurisdiction_admin',
      name: 'Joe',
      email: 'test@email.org',
      jurisdictions: [
        {
          id: 'jurisdiction-id-1',
          name: 'Jurisdiction One',
          election: {
            id: '1',
            auditName: 'audit one',
            electionName: 'election one',
            state: 'AL',
          },
        },
        {
          id: 'jurisdiction-id-2',
          name: 'Jurisdiction Two',
          election: {
            id: '2',
            auditName: 'audit two',
            electionName: 'election two',
            state: 'AL',
          },
        },
        {
          id: 'jurisdiction-id-3',
          name: 'Jurisdiction Three',
          election: {
            id: '1',
            auditName: 'audit one',
            electionName: 'election one',
            state: 'AL',
          },
        },
      ],
      organizations: [],
    },
  },
  getRounds: {
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/round',
    response: { rounds: [] },
  },
  getBallotManifestFile: (response: IFileInfo) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/ballot-manifest',
    response,
  }),
  getBatchTalliesFile: (response: IFileInfo) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/batch-tallies',
    response,
  }),
  getCVRSfile: (response: IFileInfo) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/cvrs',
    response,
  }),
  getSettings: (response: IAuditSettings) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/settings',
    response,
  }),
  putManifest: {
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/ballot-manifest',
    options: {
      method: 'PUT',
      body: manifestFormData,
    },
    response: { status: 'ok' },
  },
  putTallies: {
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/batch-tallies',
    options: {
      method: 'PUT',
      body: talliesFormData,
    },
    response: { status: 'ok' },
  },
  getAuditBoards: (auditBoards: IAuditBoard[]) => ({
    url:
      '/api/election/1/jurisdiction/jurisdiction-id-1/round/round-1/audit-board',
    response: { auditBoards },
  }),
  getBallots: (ballots: IBallot[]) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/round/round-1/ballots',
    response: { ballots },
  }),
}

export const aaApiCalls = {
  getUser: {
    url: '/api/me',
    response: {
      type: 'audit_admin',
      name: 'Joe',
      email: 'test@email.org',
      jurisdictions: [],
      organizations: [
        {
          id: 'org-id',
          name: 'State of California',
          elections: [],
        },
      ],
    },
  },
  getRounds: (rounds: IRound[]) => ({
    url: '/api/election/1/round',
    response: { rounds },
  }),
  getJurisdictions: {
    url: '/api/election/1/jurisdiction',
    response: {
      jurisdictions: [
        {
          id: 'jurisdiction-id-1',
          name: 'Jurisdiction One',
          ballotManifest: { file: null, processing: null },
          currentRoundStatus: null,
        },
        {
          id: 'jurisdiction-id-2',
          name: 'Jurisdiction Two',
          ballotManifest: { file: null, processing: null },
          currentRoundStatus: null,
        },
      ],
    },
  },
  getBatchJurisdictions: {
    url: '/api/election/1/jurisdiction',
    response: {
      jurisdictions: [
        {
          id: 'jurisdiction-id-1',
          name: 'Jurisdiction One',
          ballotManifest: { file: null, processing: null },
          batchTallies: { file: null, processing: null },
          currentRoundStatus: null,
        },
        {
          id: 'jurisdiction-id-2',
          name: 'Jurisdiction Two',
          ballotManifest: { file: null, processing: null },
          batchTallies: { file: null, processing: null },
          currentRoundStatus: null,
        },
      ],
    },
  },
  getJurisdictionFile: {
    url: '/api/election/1/jurisdiction/file',
    response: {
      file: {
        contents: null,
        name: 'file name',
        uploadedAt: 'a long time ago in a galaxy far far away',
      },
      processing: {
        status: FileProcessingStatus.Processed,
        error: null,
        startedAt: 'once upon a time',
        endedAt: 'and they lived happily ever after',
      },
    },
  },
  getContests: {
    url: '/api/election/1/contest',
    response: contestMocks.filledTargeted,
  },
  getSettings: (response: IAuditSettings) => ({
    url: '/api/election/1/settings',
    response,
  }),
  putSettings: {
    url: '/api/election/1/settings',
    options: {
      method: 'PUT',
      body: JSON.stringify(auditSettings.all),
      headers: { 'Content-Type': 'application/json' },
    },
    response: { status: 'ok' },
  },
  getSampleSizes: {
    url: '/api/election/1/sample-sizes',
    response: { sampleSizes: null },
  },
}
