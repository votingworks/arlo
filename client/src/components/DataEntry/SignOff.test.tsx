import React from 'react'
import { render, fireEvent, waitFor, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import SignOff from './SignOff'
import { dummyBoards } from './_mocks'

describe('Sign Off', () => {
  it('renders correctly', () => {
    const { container } = render(
      <SignOff auditBoard={dummyBoards()[0]} submitSignoff={jest.fn()} />
    )
    expect(container).toMatchSnapshot()
  })

  it('handles inputs', async () => {
    const submitSignoffMock = jest.fn()
    const { container, queryAllByLabelText, getByText } = render(
      <SignOff
        auditBoard={dummyBoards()[0]}
        submitSignoff={submitSignoffMock}
      />
    )

    const nameInputs = screen.getAllByLabelText('Full Name')
    const names = queryAllByLabelText('Full Name') as HTMLInputElement[]
    expect(names.length).toBe(2)
    expect(nameInputs).toHaveLength(2)

    userEvent.type(names[0], `John Doe`)
    userEvent.type(names[1], `Jane Doe`)

    const nextButton = getByText('Sign Off')
    fireEvent.click(nextButton, { bubbles: true })
    await waitFor(() => {
      expect(submitSignoffMock).toBeCalledTimes(1)
    })
    expect(container).toMatchSnapshot()
  })
})
