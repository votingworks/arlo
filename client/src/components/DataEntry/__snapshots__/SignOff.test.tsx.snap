// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Off handles inputs 1`] = `
<div>
  <div>
    <h1
      class="bp3-heading"
    >
      Audit Board #1
      : Board Member Sign-off
    </h1>
    <p>
      Thank you for completing the ballots assigned to your Audit Board. Please sign your name below to confirm that all ballots were audited to the best of your ability and in accordance with the appropriate state statutes and guidelines.
    </p>
    <p>
      If another round of auditing is needed, you will be notified by election officials.
    </p>
    <form>
      <div
        class="sc-bdVaJa kKNzTG"
      >
        <h5
          class="bp3-heading"
        >
          Audit Board Member: John Doe
        </h5>
        <label
          class="sc-gzVnrw cVHFGk"
          for="[0]"
        >
          Full Name
        </label>
        <input
          class="sc-htoDjs jNCbsf"
          id="[0]"
          name="[0]"
          value="John Doe"
        />
      </div>
      <div
        class="sc-bdVaJa kKNzTG"
      >
        <h5
          class="bp3-heading"
        >
          Audit Board Member: Jane Doe
        </h5>
        <label
          class="sc-gzVnrw cVHFGk"
          for="[1]"
        >
          Full Name
        </label>
        <input
          class="sc-htoDjs jNCbsf"
          id="[1]"
          name="[1]"
          value="Jane Doe"
        />
      </div>
      <button
        class="bp3-button bp3-disabled bp3-loading bp3-intent-primary sc-iwsKbI drSRDt"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <div
          class="bp3-spinner bp3-button-spinner"
        >
          <div
            class="bp3-spinner-animation"
          >
            <svg
              height="20"
              stroke-width="16.00"
              viewBox="-3.00 -3.00 106.00 106.00"
              width="20"
            >
              <path
                class="bp3-spinner-track"
                d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
              />
              <path
                class="bp3-spinner-head"
                d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                pathLength="280"
                stroke-dasharray="280 280"
                stroke-dashoffset="210"
              />
            </svg>
          </div>
        </div>
        <span
          class="bp3-button-text"
        >
          Sign Off
        </span>
      </button>
    </form>
  </div>
</div>
`;

exports[`Sign Off renders correctly 1`] = `
<div>
  <div>
    <h1
      class="bp3-heading"
    >
      Audit Board #1
      : Board Member Sign-off
    </h1>
    <p>
      Thank you for completing the ballots assigned to your Audit Board. Please sign your name below to confirm that all ballots were audited to the best of your ability and in accordance with the appropriate state statutes and guidelines.
    </p>
    <p>
      If another round of auditing is needed, you will be notified by election officials.
    </p>
    <form>
      <div
        class="sc-bdVaJa kKNzTG"
      >
        <h5
          class="bp3-heading"
        >
          Audit Board Member: John Doe
        </h5>
        <label
          class="sc-gzVnrw cVHFGk"
          for="[0]"
        >
          Full Name
        </label>
        <input
          class="sc-htoDjs jNCbsf"
          id="[0]"
          name="[0]"
          value=""
        />
      </div>
      <div
        class="sc-bdVaJa kKNzTG"
      >
        <h5
          class="bp3-heading"
        >
          Audit Board Member: Jane Doe
        </h5>
        <label
          class="sc-gzVnrw cVHFGk"
          for="[1]"
        >
          Full Name
        </label>
        <input
          class="sc-htoDjs jNCbsf"
          id="[1]"
          name="[1]"
          value=""
        />
      </div>
      <button
        class="bp3-button bp3-disabled bp3-intent-primary sc-iwsKbI drSRDt"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          Sign Off
        </span>
      </button>
    </form>
  </div>
</div>
`;
